presto-testing/src/main/java/io/prestosql/testing/AbstractTestIntegrationSmokeTest.java

```diff
@@ -203,21 +207,26 @@ public void testSelectAll()
     /**
      * Test interactions between optimizer (including CBO), scheduling and connector metadata APIs.
      */
-    @Test(timeOut = 300_000)
-    public void testJoinWithEmptySides()
+    @Test(timeOut = 300_000, dataProvider = "joinDistributionTypes")
+    public void testJoinWithEmptySides(JoinDistributionType joinDistributionType)
+    {
+        Session session = Session.builder(getSession())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, joinDistributionType.toString())
+                .setSystemProperty(JOIN_REORDERING_STRATEGY, NONE.toString())
+                .build();
```

Comment:
Maybe move `io.prestosql.testing.AbstractTestJoinQueries#noJoinReordering` to ATQF

Reply:
Let me do it in separate pull request

Reply:
https://github.com/prestosql/presto/pull/4987